packages/language-service/src/lib/service.ts

```diff
@@ -295,41 +294,63 @@ export class StylableLanguageService {
+                groupedCompletions.set(lspCompletion.label, lspCompletion);
+            } else {
+                continue;
+            }
+
+            const textEdit: TextEdit = TextEdit.replace(
+                stComp.range
+                    ? stComp.range
+                    : new ProviderRange(
+                          new ProviderPosition(position.line, Math.max(position.character - 1, 0)),
+                          position
+                      ),
+                typeof stComp.insertText === 'string' ? stComp.insertText : stComp.insertText.source
+            );
+            lspCompletion.insertTextFormat = 2;
+            lspCompletion.detail = stComp.detail;
+            lspCompletion.textEdit = textEdit;
+            // override sorting order to the top
+            // todo: decide on actual sorting for all stylable completions
+            lspCompletion.sortText = 'a';
+            lspCompletion.filterText =
+                typeof stComp.insertText === 'string'
+                    ? stComp.insertText
+                    : stComp.insertText.source;
+            if (stComp.additionalCompletions) {
+                lspCompletion.command = Command.create(
+                    'additional',
+                    'editor.action.triggerSuggest'
                 );
-                lspCompletion.insertTextFormat = 2;
-                lspCompletion.detail = com.detail;
-                lspCompletion.textEdit = ted;
-                lspCompletion.sortText = com.sortText;
-                lspCompletion.filterText =
-                    typeof com.insertText === 'string' ? com.insertText : com.insertText.source;
-                if (com.additionalCompletions) {
-                    lspCompletion.command = Command.create(
-                        'additional',
-                        'editor.action.triggerSuggest'
-                    );
-                } else if (com.triggerSignature) {
-                    lspCompletion.command = Command.create(
-                        'additional',
-                        'editor.action.triggerParameterHints'
-                    );
-                }
-                return lspCompletion;
-            })
-            .concat(this.cssService.getCompletions(cleanDocument, position));
+            } else if (stComp.triggerSignature) {
+                lspCompletion.command = Command.create(
+                    'additional',
+                    'editor.action.triggerParameterHints'
+                );
+            }
+        }
+
+        const cssCompletions = this.cssService.getCompletions(cleanDocument, position);
+
+        for (const cssComp of cssCompletions) {
+            const label = cssComp.label;
+
+            if (!groupedCompletions.has(label)) {
+                // CSS declaration property names have built in sorting
+                // at-rules, rules and declaration values do not
+                cssComp.sortText = cssComp.sortText || 'z';
```

Comment:
Does this prevent compilation to be sorted by the label?

Reply:
No, we still beat it in the sort - this maintains their internal sorting.